### PR TITLE
CORE: free parser in lib destructor

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**

### CODE REVIEW

File: src/core/ucc_constructor.c [hunk 1]
```diff
@@ -79,7 +79,8 @@ ucc_status_t ucc_constructor(void)
 
 __attribute__((destructor)) static void ucc_destructor(void)
 {
-    if (ucc_global_config.component_path_default) {
-        ucc_free(ucc_global_config.component_path_default);
```
Comment (vspetrov) [inline src/core/ucc_constructor.c:83]: why this ucc_free is removed?

Comment (Sergei-Lebedev) [reply src/core/ucc_constructor.c:83]: ucs config parser release iterates over all fields declared in config table and calls corresponding destructor. String field destructor is here https://github.com/openucx/ucx/blob/1c883359c954e6678e0a2dbfea4efbbedb029d60/src/ucs/config/parser.c#L105
It calls free for pointer no matter if it was parsed (allocated) or not.

Comment (vspetrov) [reply src/core/ucc_constructor.c:83]: but "component_path_default" is NOT declared in the config table. It is detected automatically. "component_path" is declared. am i missing smth?

Comment (Sergei-Lebedev) [reply src/core/ucc_constructor.c:83]: yes, component path default is not part of config table, but if component path is not provided by user then component path pointer assigned to component path default
https://github.com/openucx/ucc/blob/dfdfb594da7bd61af7cb2b74856b9e077c3b1bfd/src/core/ucc_constructor.c#L37

Comment (vspetrov) [reply src/core/ucc_constructor.c:83]: got it

**Review by vspetrov — APPROVED** [review]

**Review by manjugv — APPROVED** [review]

**Pull request merged**